Src/VimMac/VimHost.cs

```diff
@@ -702,7 +701,18 @@ public bool SaveTextAs(string text, string filePath)
 
         public bool ShouldCreateVimBuffer(ITextView textView)
         {
-            return !textView.TextBuffer.ContentType.IsDebuggerWindow();
+            var shouldCreateVimBuffer = !textView.TextBuffer.ContentType.IsDebuggerWindow();
+
+            if (shouldCreateVimBuffer)
+                textView.Closed += TextView_Closed;
+
+            return shouldCreateVimBuffer;
+
+            void TextView_Closed(object sender, EventArgs e)
+            {
+                _vim.RemoveVimBuffer(textView);
```

Comment:
I was puzzled at first why you needed to make this change. Had to dig a bit to figure out that this logic in the Windows version is being handled above the shared layer. 

https://github.com/VsVim/VsVim/blob/master/Src/VsVimShared/HostFactory.cs#L201

That implementation should be moved down to the shared layer at some point certainly. 

Rather than calling `_vim.RemoveBuffer` I think you should do essentially

```cs
if (_vim.TryGetBuffer(textView, out var vimBuffer))
   vimBuffer.Close()
```

Closing the `IVimBuffer` frees up other items in addition to removing from the buffer list. Looking now I don't think I should have made `IVimBuffer.RemoveBuffer` a public API. Or at least it should be making sure that `IVimBuffer` is closed.

Reply:
Ok. Thanks. 

It looks like I could move this part `textView.Closed += (x, y) => vimBuffer.Close()` into the VimBuffer constructor